src/cryptsetup/cryptsetup.c

```diff
@@ -38,7 +38,15 @@
 #include "strv.h"
 #include "util.h"
 
-static const char *arg_type = NULL; /* CRYPT_LUKS1, CRYPT_TCRYPT or CRYPT_PLAIN */
+/* libcryptsetup define for any LUKS version, compatible with libcryptsetup 1.x */
+#ifndef CRYPT_LUKS
+#define CRYPT_LUKS NULL
+#endif
```

Comment:
hmm, not too keen to keep such compat kludges in place if we don't have to. How universal is libcryptsetup 2.x adoption at this time? If it's sufficiently widely used I'd prfer to just bump the dependency version and then avoid and compatibility kludges like this and just pass NULL literally wherever neded

Reply:
It's a release candidate so it's not released as 'stable' yet and therefore the adoption level is far from being widely used.

Anyway, feel free to replace (or should I update PR?) it with crypt_load(cd, NULL, NULL). As I wrote in comments, the NULL type is interpreted as "load any luks version" even by libcryptsetup 1.x.

Reply:
uhhh, we usually don't merge patches that make use of features of yet unreleased library dependencies. Unless something is really released who knows whether the API is still changed? what's the timeframe for releasing this for good?

Reply:
API will not change in this phase. The note about format is meant for internal metadata (LUKS2 uses JSON). So it is safe to merge this - but if you prefer do this once libcryptsetup2 is stable release, no problem with it (just cryptsetup-generator will not be able to activate LUKS2 devices.)